src/main/java/com/internet/shop/controllers/RegistrationController.java

```diff
@@ -32,10 +35,14 @@ protected void doPost(HttpServletRequest req, HttpServletResponse resp)
         String pswRepeat = req.getParameter("psw-repeat");
         if (psw.equals(pswRepeat)) {
             User user = new User(name, login, psw);
+            Set<Role> roleSet = new HashSet<>();
+            roleSet.add(Role.of("USER"));
+            user.setRoles(roleSet);
             ShoppingCart shoppingCart = new ShoppingCart();
             user = userService.create(user);
-            shoppingCart.setUserId(user.getId());
             shoppingCartService.create(shoppingCart);
```

Comment:
let's create a shopping cart when userId is already set

Reply:
fixed